stmtcacher.go

```diff
@@ -66,6 +81,26 @@ func (sc *stmtCacher) QueryRow(query string, args ...interface{}) RowScanner {
 	return stmt.QueryRow(args...)
 }
 
+func (sc *stmtCacher) Clear() error {
+	sc.mu.Lock()
+	defer sc.mu.Unlock()
+
+	for key, stmt := range sc.cache {
+		delete(sc.cache, key)
```

Comment:
Would love thoughts on this. Would we want to delete it in all cases? or perhaps leave it when `err != nil`?

Reply:
Delete it in all cases. `Stmt.Close` doesn't say that an error leaves the Stmt valid, so it should be assumed invalid and not left to cause more problems.

Reply:
💯 that was my thinking. Didn't want to leave around something which just repeatedly caused errors on calls to Clear().